plugins/permission-backend/src/service/router.ts

```diff
@@ -93,28 +102,51 @@ const evaluatePermissionRequestBatchSchema: z.ZodSchema<EvaluatePermissionReques
  */
 export interface RouterOptions {
   logger: Logger;
-  discovery: PluginEndpointDiscovery;
+  discovery: DiscoveryService;
   policy: PermissionPolicy;
-  identity: IdentityApi;
+  identity?: IdentityApi;
   config: Config;
+  auth?: AuthService;
+  httpAuth?: HttpAuthService;
+  userInfo?: UserInfoService;
 }
 
 const handleRequest = async (
   requests: IdentifiedPermissionMessage<EvaluatePermissionRequest>[],
-  user: BackstageIdentityResponse | undefined,
   policy: PermissionPolicy,
   permissionIntegrationClient: PermissionIntegrationClient,
-  authHeader?: string,
+  credentials: BackstageCredentials<
+    BackstageNonePrincipal | BackstageUserPrincipal
+  >,
+  auth: AuthService,
+  userInfo: UserInfoService,
 ): Promise<IdentifiedPermissionMessage<EvaluatePermissionResponse>[]> => {
   const applyConditionsLoaderFor = memoize((pluginId: string) => {
     return new DataLoader<
       ApplyConditionsRequestEntry,
       ApplyConditionsResponseEntry
     >(batch =>
-      permissionIntegrationClient.applyConditions(pluginId, batch, authHeader),
+      permissionIntegrationClient.applyConditions(pluginId, credentials, batch),
     );
   });
 
+  let user: BackstageIdentityResponse | undefined;
+  if (auth.isPrincipal(credentials, 'user')) {
+    const { ownershipEntityRefs } = await userInfo.getUserInfo(credentials);
+    const { token } = await auth.getPluginRequestToken({
+      onBehalfOf: credentials,
+      targetPluginId: 'catalog', // TODO: unknown at this point
+    });
+    user = {
+      identity: {
+        type: 'user',
+        userEntityRef: credentials.principal.userEntityRef,
+        ownershipEntityRefs,
+      },
+      token,
```

Comment:
We'll want to start forwarding the credentials here as well right? Allow users to switch from using the token to the new credentials

Reply:
Never mind, let's not add credentials here unless needed. Deprecating the `token` would be the first step, if it turns out that credentials are needed we can add that back in later